### DISCUSSION TITLE
Mix of typescript and plain JS?

### DISCUSSION THREAD

**Discussion by mindplay-dk** [danny-avila/LibreChat#851] [Q&A] [answered]
Just wondering, since the codebase appears to be a mix of TypeScript and plain JavaScript - is the project currently moving from one to the other, or is it a deliberate choice to mix languages?

A search for "typescript" in either issues or discussions did not give any results - the word "typescript" does not occur in the contributor guidelines either. Whether you're currently moving from one to the other, or mixing for some reason, the contributor guide should probably explain which to use, and when, and why. 🙂

**Comment by danny-avila** [comment, marked as answer]
Thanks for asking. I can add this in.

the project was originally, entirely JS.

The frontend is being converted to TS, nearly complete. This can be done because React allows the mixture of the two before compiling the code, and it’s the norm to compile/bundle the code.

The backend would be more involved because it would have to be done in “one go,” unless we add some compiling for the server, which is not the norm and would add complexity. This is lower in priority and may not happen.

**Reply by danny-avila** [reply]
updated docs: https://github.com/danny-avila/LibreChat/blob/main/.github/CONTRIBUTING.md
